patches/minecraft/net/minecraft/client/util/RecipeBookCategories.java.patch

```diff
@@ -0,0 +1,151 @@
+--- a/net/minecraft/client/util/RecipeBookCategories.java
++++ b/net/minecraft/client/util/RecipeBookCategories.java
+@@ -9,23 +9,24 @@
+ import net.minecraftforge.api.distmarker.OnlyIn;
+ 
+ @OnlyIn(Dist.CLIENT)
+-public enum RecipeBookCategories {
+-   SEARCH(new ItemStack(Items.field_151111_aL)),
+-   BUILDING_BLOCKS(new ItemStack(Blocks.field_196584_bK)),
+-   REDSTONE(new ItemStack(Items.field_151137_ax)),
+-   EQUIPMENT(new ItemStack(Items.field_151036_c), new ItemStack(Items.field_151010_B)),
+-   MISC(new ItemStack(Items.field_151129_at), new ItemStack(Items.field_151034_e)),
+-   FURNACE_SEARCH(new ItemStack(Items.field_151111_aL)),
+-   FURNACE_FOOD(new ItemStack(Items.field_151147_al)),
+-   FURNACE_BLOCKS(new ItemStack(Blocks.field_150348_b)),
+-   FURNACE_MISC(new ItemStack(Items.field_151129_at), new ItemStack(Items.field_151166_bC)),
+-   BLAST_FURNACE_SEARCH(new ItemStack(Items.field_151111_aL)),
+-   BLAST_FURNACE_BLOCKS(new ItemStack(Blocks.field_150450_ax)),
+-   BLAST_FURNACE_MISC(new ItemStack(Items.field_151037_a), new ItemStack(Items.field_151149_ai)),
+-   SMOKER_SEARCH(new ItemStack(Items.field_151111_aL)),
+-   SMOKER_FOOD(new ItemStack(Items.field_151147_al)),
+-   STONECUTTER(new ItemStack(Items.field_221782_da)),
+-   CAMPFIRE(new ItemStack(Items.field_151147_al));
++@net.minecraftforge.registries.ObjectHolder("minecraft")
++public class RecipeBookCategories extends net.minecraftforge.registries.ForgeRegistryEntry<RecipeBookCategories> {
```

Comment:
This breaks binary compatibility

Reply:
Any idea how to do it without breaking compatibility?

Reply:
Fixed ABI